scripts/solve_network.py

```diff
@@ -223,33 +229,62 @@ def add_CCL_constraints(n, config):
         "Adding per carrier generation capacity constraints for " "individual countries"
     )
 
-    gen_country = n.generators.bus.map(n.buses.country)
     capacity_variable = n.model["Generator-p_nom"]
 
-    lhs = []
+    # get carriers to which CCL constraints apply
+    ccl_carriers = agg_p_nom_minmax.columns.get_level_values(1).unique()
     ext_carriers = n.generators.query("p_nom_extendable").carrier.unique()
-    for c in ext_carriers:
-        ext_carrier = n.generators.query("p_nom_extendable and carrier == @c")
+    ccl_carriers = ccl_carriers[ccl_carriers.isin(ext_carriers)]
+
+    # If no CCL carriers found, return early
+    if not ccl_carriers.any():
+        logger.info(
+            "No CCL carriers found that are extendable. Skipping CCL constraints."
+        )
+        return
+
+    for c in ccl_carriers:
+        # Handle extendable capacities
+        ccl_carrier = n.generators[n.generators.carrier == c]
         country_grouper = (
-            ext_carrier.bus.map(n.buses.country)
+            ccl_carrier.bus.map(n.buses.country)
             .rename_axis("Generator-ext")
             .rename("country")
         )
-        ext_carrier_per_country = capacity_variable.loc[
-            country_grouper.index
-        ].groupby_sum(country_grouper)
-        lhs.append(ext_carrier_per_country)
-    lhs = merge(lhs, dim=pd.Index(ext_carriers, name="carrier"))
+        lhs = (
+            capacity_variable.loc[country_grouper.index].groupby(country_grouper).sum()
+        )

```

Comment:
Just to be sure: if country isn't included into `agg_p_nom_minmax.csv`, it isn't a part of `country_grouper` and will not be included into the constraint formulation, right?